### DISCUSSION TITLE
Is there a way to set different upstreams for routes under the same service?

### DISCUSSION THREAD

**Discussion by jianmaikj** [Kong/kong#8857] [Help]
Is there a way to set different upstreams for routes under the same service?
thanks！

**Comment by ms2008** [comment]
Sounds similar to this [Route By Header](https://docs.konghq.com/hub/kong-inc/route-by-header/), but for the route.

**Reply by jianmaikj** [reply]
> Sounds similar to this [Route By Header](https://docs.konghq.com/hub/kong-inc/route-by-header/), but for the route.

Yeah, it is useful, thank you.but it only for enterprise version, is there an open source solution?

**Comment by jschmid1** [comment]
> Sounds similar to this [Route By Header](https://docs.konghq.com/hub/kong-inc/route-by-header/), but for the route.

If you want to route based on the request context, this would be an option. 

> Is there a way to set different upstreams for routes under the same service?
thanks！

Would it be an option to set-up multiple services (each with their respective upstream) and point routes against those?
